sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/V1Writes.scala

```diff
@@ -102,8 +102,8 @@ object V1Writes extends Rule[LogicalPlan] with SQLConfHelper {
     val requiredOrdering = write.requiredOrdering.map(_.transform {
       case a: Attribute => attrMap.getOrElse(a, a)
     }.asInstanceOf[SortOrder])
-    val outputOrdering = query.outputOrdering
-    val orderingMatched = isOrderingMatched(requiredOrdering, outputOrdering)
+    val outputOrdering = empty2NullPlan.outputOrdering
```

Comment:
I rough remember before we did not support preserve ordering through empty2null so use the `query.outputOrdering`. I think use `empty2NullPlan.outputOrdering` is the expected behavior

Reply:
Yea Project is an `OrderPreservingUnaryNode` so it should be fine.